plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTransactionsTable.java

```diff
@@ -0,0 +1,213 @@
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.predicate.Range;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.type.TypeManager;
+import io.trino.spi.type.TypeSignature;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static com.google.common.collect.MoreCollectors.onlyElement;
+import static io.airlift.json.JsonCodec.listJsonCodec;
+import static io.trino.plugin.deltalake.DeltaLakeErrorCode.DELTA_LAKE_INVALID_SCHEMA;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogDir;
+import static io.trino.plugin.deltalake.transactionlog.checkpoint.TransactionLogTail.getEntriesFromJson;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.StandardTypes.JSON;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakeTransactionsTable
+        implements SystemTable
+{
+    private static final JsonCodec<List<DeltaLakeTransactionLogEntry>> TRANSACTION_LOG_ENTRIES_CODEC = listJsonCodec(DeltaLakeTransactionLogEntry.class);
+
+    private final SchemaTableName tableName;
+    private final String tableLocation;
+    private final TrinoFileSystemFactory fileSystemFactory;
+    private final TransactionLogAccess transactionLogAccess;
+    private final ConnectorTableMetadata tableMetadata;
+
+    public DeltaLakeTransactionsTable(
+            SchemaTableName tableName,
+            String tableLocation,
+            TrinoFileSystemFactory fileSystemFactory,
+            TransactionLogAccess transactionLogAccess,
+            TypeManager typeManager)
+    {
+        requireNonNull(typeManager, "typeManager is null");
+        this.tableName = requireNonNull(tableName, "tableName is null");
+        this.tableLocation = requireNonNull(tableLocation, "tableLocation is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.transactionLogAccess = requireNonNull(transactionLogAccess, "transactionLogAccess is null");
+
+        this.tableMetadata = new ConnectorTableMetadata(
+                requireNonNull(tableName, "tableName is null"),
+                ImmutableList.<ColumnMetadata>builder()
+                        .add(new ColumnMetadata("version", BIGINT))
+                        .add(new ColumnMetadata("transaction", typeManager.getType(new TypeSignature(JSON))))
+                        .build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
```

Comment:
There is much duplication between this method and `DeltaLakeHistoryTable:: pageSource` can you extract common parts to some utility methods, or some hierarchy?